src/main/scala/com/campudus/tableaux/TableauxRouter.scala

```diff
@@ -33,16 +34,25 @@ class TableauxRouter(verticle: Starter) extends Router with VertxAccess {
     case Get(tableIdColumnsIdRowsId(tableId, columnId, rowId)) => getAsyncReply(controller.getCell(tableId.toLong, columnId.toLong, rowId.toLong))
     case Post("/reset") => getAsyncReply(controller.resetDB())
     case Post("/tables") => getAsyncReply {
-      getJson(req) flatMap { json => controller.createTable(json.getString("tableName")) }
+      import scala.collection.JavaConverters._
+      getJson(req) flatMap { json =>
+        if (json.getFieldNames.asScala.toSeq.contains("cols")) {
+          if (json.getFieldNames.asScala.toSeq.contains("rows")) {

```

Comment:
I'd put `json.getFieldNames.asScala.toSeq` into a variable as you call it twice.